code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/sassysage
+
+/obj/item/food/pizza/sassysage/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/sassysage, 6, 30)
+
+/obj/item/food/pizza/sassysage/raw
+	name = "raw sassysage pizza"
+	icon_state = "sassysagepizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | MEAT | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/sassysage/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/sassysage, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice/sassysage
+	name = "sassysage pizza slice"
+	desc = "Deliciously sassy."
+	icon_state = "sassysagepizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/pineapple
+	name = "\improper Hawaiian pizza"
+	desc = "The pizza equivalent of Einstein's riddle."
+	icon_state = "pineapplepizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 20, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 6, /datum/reagent/consumable/pineapplejuice = 8)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pineapple" = 2, "ham" = 2)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT | FRUIT | PINEAPPLE
+	slice_type = /obj/item/food/pizzaslice/pineapple
+
+/obj/item/food/pizza/pineapple/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/pineapple, 6, 30)
+
+/obj/item/food/pizza/pineapple/raw
+	name = "raw Hawaiian pizza"
+	icon_state = "pineapplepizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | MEAT | FRUIT | PINEAPPLE | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/pineapple/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/pineapple, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice/pineapple
+	name = "\improper Hawaiian pizza slice"
+	desc = "A slice of delicious controversy."
+	icon_state = "pineapplepizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pineapple" = 2, "ham" = 2)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT | FRUIT | PINEAPPLE
+
+/obj/item/food/pizza/arnold
+	name = "\improper Arnold pizza"
+	desc = "Hello, you've reached Arnold's pizza shop. I'm not here now, I'm out killing pepperoni."
+	icon_state = "arnoldpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/nutriment/vitamin = 6, /datum/reagent/iron = 10, /datum/reagent/medicine/omnizine = 30)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pepperoni" = 2, "9 millimeter bullets" = 2)
+	slice_type = /obj/item/food/pizzaslice/arnold
+
+/obj/item/food/pizza/arnold/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.